### PR TITLE
chore: add debug logs for session logic

### PR DESCRIPTION
**Pull request by falconandy**
### Summary

1. Removes some logic from `SessionManager.setSessionId`/`SessionManager.getSessionId`. If the logic is required in some cases, please let me know.
2. Adds some logging to try to debug session-logic issues. `Debug` level isn't used - too much output for other debug messages.

### Checklist

* [x] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-TypeScript/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  No

### CODE REVIEW

File: packages/analytics-client-common/src/session-manager.ts [hunk 1]
```diff
@@ -4,7 +4,6 @@ import { getCookieName as getStorageKey } from './cookie-name';
 export class SessionManager implements ISessionManager {
   storageKey: string;
   cache: UserSession;
-  isSessionCacheValid = true;
```
Comment (qingzhuozhen) [inline packages/analytics-client-common/src/session-manager.ts:7]: why remove this? cc @kevinpagtakhan do you know if this one is safe to remove?

Comment (falconandy) [reply packages/analytics-client-common/src/session-manager.ts:7]: `isSessionCacheValid`  was used only in `getSessionId`/`setSessionId` logic and not used in others (`getDeviceId`/`setDeviceId`, `getUserId`/`setUserId`, `getLastEventTime`/`setLastEventTime`, etc)

Comment (qingzhuozhen) [reply packages/analytics-client-common/src/session-manager.ts:7]: Maybe we can keep it if it doesn't hurt the debugging purpose, or you can move forward if you believe this doesn't break anything

**Review by qingzhuozhen — APPROVED** [review]

File: packages/analytics-client-common/src/session-manager.ts (hunk 1 above)
Comment (falconandy) [reply packages/analytics-client-common/src/session-manager.ts:7]: I believe it shouldn't break anytihng

**Pull request merged**